ddterm/app/application.js

```diff
@@ -650,7 +655,9 @@ class Application extends Gtk.Application {
     }
 
     restore_session() {
-        const [, data] = GLib.file_get_contents(this.session_file_path);
+        const [ok, data] = GLib.file_get_contents(this.session_file_path);
+        if (!ok)
+            GLib.unlink(this.session_file_path);
```

Comment:
This will never happen. `ok` will be always `true` here. Instead, `file_get_contents()` will `throw` if there's an I/O error. Also, I'm not sure if it's a good idea to remove the file on I/O error.

Instead, the file should be removed if its content is invalid - i.e. if it can't be parsed as `GLib.Variant` or if something goes wrong when you try to restore the session from it.

Reply:
removed this